include/xtensor-io/xhighfive.hpp

```diff
@@ -0,0 +1,666 @@
+#include <vector>
+
+#include <highfive/H5DataSet.hpp>
+#include <highfive/H5DataType.hpp>
+#include <highfive/H5DataSpace.hpp>
+#include <highfive/H5File.hpp>
+
+#include <xtensor/xarray.hpp>
+#include <xtensor/xtensor.hpp>
+
+#include "xtensor-io.hpp"
+
+namespace xt
+{
+    namespace extensions
+    {
+        /**
+        * Check if a path exists (is a Group or DataSet) in an open HDF5 file.
+        *
+        * @param file opened HighFive::File
+        * @param path path of the Group/DataSet
+        */
+        inline bool exist(const HighFive::File &file, const std::string &path)
+        {
+            // find first "/"
+            std::size_t idx = path.find("/");
+
+            // loop over all groups
+            while (true)
+            {
+                // terminate if all "/" have been found
+                if (std::string::npos == idx)
+                {
+                    break;
+                }
+
+                // create group if needed
+                if (idx > 0)
+                {
+                    // get group name
+                    std::string name(path.substr(0,idx));
+                    // create if needed
+                    if (!file.exist(name))
+                    {
+                        return false;
+                    }
+                }
+
+                // proceed to next "/"
+                idx = path.find("/",idx+1);
+            }
+
+            return file.exist(path);
+        }
+
+        /**
+        * Recursively create groups in an open HDF5 file such that a DataSet can be created.
+        * For example if the path = "/path/to/dataset", this function will create the groups
+        * "/path" and "/path/to".
+        *
+        * @param file opened HighFive::File
+        * @param path path of the DataSet
+        */
+        inline void createGroup(HighFive::File &file, const std::string &path)
+        {
+            // find first "/"
+            std::size_t idx = path.find("/");
+
+            // loop over all groups
+            while (true)
+            {
+                // terminate if all "/" have been found
+                if (std::string::npos == idx)
+                {
+                    return;
+                }
+
+                // create group if needed
+                if (idx > 0)
+                {
+                    // get group name
+                    std::string name(path.substr(0,idx));
+
+                    // create if needed
+                    if (!file.exist(name))
+                    {
+                        file.createGroup(name);
+                    }
+                }
+
+                // proceed to next "/"
+                idx = path.find("/",idx+1);
+            }
+        }
+    }
+
+    namespace detail
+    {
+        inline auto highfive_file_mode(xt::file_mode mode)
+        {
+            switch (mode)
+            {
+                case xt::file_mode::create :
+                  return HighFive::File::Create;
```

Comment:
Indentation :)
Also, personally I wouldn't do a space in front of the colon, hehe. But this is turbo-nitpicking.

Reply:
Will do.